js/main.js

```diff
@@ -1,19 +1,119 @@
-// Функция, возвращающая случайное число
-const getRandomInteger = function (min, max) {
-  if (min >= 0 && max >= 0) {
-    return Math.floor(Math.random() * (max - min + 1) + min);
+const AUTHORS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const TYPES = ['palace', 'flat', 'house', 'bungalow', 'hotel'];
+
+const CHECKIN = ['12:00', '13:00', '14:00'];
+
+const CHECKOUT = ['12:00', '13:00', '14:00'];
+
+const FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+
+const PHOTOS = ['https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/duonguyen-8LrGtIxxa4w.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg'];
+
+const SIMILAR_AD_COUNT = 10;
+
+const ADDRESS_LOCATION_X = 60;
+
+const ADDRESS_LOCATION_Y = 590;
+
+const Prices = {
+  PRICE_MIN: 500,
+  PRICE_MAX: 5000,
+};
+
+const Rooms = {
+  ROOM_MIN: 1,
+  ROOM_MAX: 10,
+};
+
+const Guests = {
+  GUESTS_MIN: 1,
+  GUESTS_MAX: 10,
+};
+
+const LocationX = {
+  LAT_X_MIN: 35.65000,
+  LAT_X_MAX: 35.70000,
+};
+
+const LocationY = {
+  LNG_Y_MIN: 35.65000,
+  LNG_Y_MAX: 35.70000,
+};
+
+const DIGIT = 5;
+
+// Функция взята из интернета и доработана
+// Источник - https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_random
+const getRandomPositiveInteger = function (a, b) {
+  const lower = Math.ceil(Math.min(Math.abs(a), Math.abs(b)));
+  const upper = Math.floor(Math.max(Math.abs(a), Math.abs(b)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
+};
+
+// Функция взята из интернета и доработана
+// Источник - https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_random
+const getRandomPositiveFloat = function (a, b, digits = 1) {
+  const lower = Math.min(Math.abs(a), Math.abs(b));
+  const upper = Math.max(Math.abs(a), Math.abs(b));
+  const result = Math.random() * (upper - lower) + lower;
+  return result.toFixed(digits);
+};
+
+// Функция, возвращаяющая рандомный элемент
+const getRandomArrayElement = function (elements) {
+  return elements[getRandomPositiveInteger(0, elements.length - 1)];
+};
+
+// Функция, возвращает массив рандомной длины
+const getRandomArrayLength = function (arr, length) {
+  const newArr = [];
+  for (let i = 0; i < length; i += 1) {
+    newArr.push(FEATURES[i]);
   }
-  return null;
+  return newArr;
 };
-getRandomInteger(-4, 56);
 
-// Функция, возвращающая случайное число с плавающей точкой
-const getRandomFloat = function (min, max, digit) {
-  if (min < 0 || max < 0) {
-    return null;
-  } else {
-    const randomNum = Math.random() * (max - min) + min;
-    return randomNum.toFixed(digit);
+// Функция, создающая строку с рандомным значением
+const getRandomConcatString = function () {
+  const newString = getRandomArrayElement(AUTHORS);
+  if (newString < AUTHORS.length) {
```

Comment:
```suggestion
  const newString = getRandomArrayElement(AUTHORS);
  
  if (newString < AUTHORS.length) {
```

больше пробелов богу пробелов